File: clipwright/api/waveform.py
"""波形 API — 生成音频波形数据。"""

from __future__ import annotations

from fastapi import APIRouter, HTTPException
from fastapi.params import Body

from clipwright.services.waveform import WaveformGenerator

router = APIRouter(prefix="/api/waveform", tags=["waveform"])


@router.post("/generate")
async def generate_waveform(
    audio_path: str = Body(...),
    samples: int = Body(default=200),
) -> dict:
    """从音频文件生成波形采样数据（用于时间轴可视化）。"""
    result = await WaveformGenerator.generate(audio_path, samples)
    return result

File: web/src/services/api/waveform.ts
import { getApiClient } from './client';

export interface WaveformData {
  samples: number[];
  duration_sec: number;
}

export const waveformApi = {
  async generate(audioPath: string, samples?: number): Promise<WaveformData> {
    const { data } = await getApiClient().post('/api/waveform/generate', {
      audio_path: audioPath,
      samples: samples ?? 200,
    });
    return data;
  },
};
